torch/export/dynamic_shapes.py

```diff
@@ -832,32 +837,32 @@ def check_symbols(path, tensor, shape):
             for i, dim in shape.items():
                 if isinstance(dim, _Dim):
                     check_same_bounds(dim)
-                elif not isinstance(dim, int) and dim is not None:
+                elif not isinstance(dim, int) and dim not in [None, True]:
                     raise UserError(
                         UserErrorType.INVALID_INPUT,
                         f"Unexpected dimension mapped to index {i} in input tensor shape {shape} "
                         f"specified at `dynamic_shapes{keystr(path)}` "
-                        f"(expected None, an int, or a Dim, but got {dim} instead)",
+                        f"(expected None, True, an int, or a Dim, but got {dim} instead)",
                         case_name="dynamic_shapes_validation",
                     )
         elif isinstance(shape, (tuple, list)):
             for i, dim in enumerate(shape):
                 if isinstance(dim, _Dim):
                     check_same_bounds(dim)
-                elif not isinstance(dim, int) and dim is not None:
+                elif not isinstance(dim, int) and dim not in [None, True]:
                     raise UserError(
                         UserErrorType.INVALID_INPUT,
                         f"Unexpected dimension #{i} in input tensor shape {shape} "
                         f"specified at `dynamic_shapes{keystr(path)}` "
-                        f"(expected None, an int, or a Dim, but got {dim} instead)",
+                        f"(expected None, True, an int, or a Dim, but got {dim} instead)",
                         case_name="dynamic_shapes_validation",
                     )
-        elif shape is not None:
+        elif shape not in [None, True]:
```

Comment:
Likewise, I don't think it should ever be `True` here, unless you are thinking of a different role for `True` here that applies to subtrees.

Reply:
Yeah I mentioned above, this allows True to apply to the entire structure, a tensor, or a Dim